Kitodo-LongTimePreservationValidationModule/src/main/java/org/kitodo/longtimepreservationvalidationmodule/KitodoOutputHandler.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.longtimepreservationvalidationmodule;
+
+import edu.harvard.hul.ois.jhove.App;
+import edu.harvard.hul.ois.jhove.HandlerBase;
+import edu.harvard.hul.ois.jhove.Message;
+import edu.harvard.hul.ois.jhove.Module;
+import edu.harvard.hul.ois.jhove.OutputHandler;
+import edu.harvard.hul.ois.jhove.RepInfo;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.kitodo.api.validation.State;
+import org.kitodo.api.validation.ValidationResult;
+
+public class KitodoOutputHandler extends HandlerBase {
+
+    /**
+     * The numbers are used to initialize a calendar object which is never used
+     * later, Hoverer, the array must be in place or a
+     * {@link NullPointerException} will be thrown when JHove tries to access
+     * the array members to initialize the unused object. The Java epoch is
+     * defined in milliseconds since 1970-01-01, so these numbers practically
+     * mean <i>zero</i>.
+     */
+    private static final int[] EPOCH_ZERO = new int[] {1970, 1, 1 };
+
+    private static final Pattern LINE_SPLITTER = Pattern.compile(System.lineSeparator());
+
+    private static final Constructor<ValidationResult> VALIDATION_RESULT_CONSTRUCTOR;
+
+    static {
+        try {
+            VALIDATION_RESULT_CONSTRUCTOR = ValidationResult.class.getDeclaredConstructor(State.class,
+                Collection.class);
+            VALIDATION_RESULT_CONSTRUCTOR.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            throw new UndeclaredThrowableException(e);
+        }
+    }
+
+    /**
+     * Method that is executed before the constructor of the parent class.
+     *
+     * @return {@code null}. The value is not used.
+     */
+    private static final String beforeParentConstructor() {
+        Logger.getLogger("edu.harvard.hul.ois.jhove.handler").setLevel(Level.OFF);
+        return null;
+    }
+
+    private final List<String> messages = new ArrayList<String>();
+
+    private ValidationResultState state;
+
+    protected KitodoOutputHandler() {
+        super(beforeParentConstructor(), null, EPOCH_ZERO, null, null);
+    }
+
+    private void addMessage(Message message) {
+        StringBuilder line = new StringBuilder(127);
+        line.append(message.getMessage());
+        String subMessage = message.getSubMessage();
+        if (Objects.nonNull(subMessage = null)) {
```

Comment:
You are assigning a  value before checking its content with `null` - I do not think, that you want this. Maybe you mean `if (Objects.nonNull(subMessage)) {`.